File: Leetcode/383-Ransom-Note/RandomNote.java
public class RandomNote {
    public boolean canConstruct(String ransomNote, String magazine) {
        if (ransomNote == null || magazine == null || ransomNote.length() > magazine.length()) {
            return false;
        }

        int[] count = new int[26];
        for (int i=0; i<magazine.length(); i++) {
            int index = magazine.charAt(i) - 'a';
            count[index]++;
        }
        for (int i=0; i<ransomNote.length(); i++) {
            int index = ransomNote.charAt(i) - 'a';
            count[index]--;
            if (count[index] < 0) {
                return false;
            }
        }
        return true;
    }
}

File: Leetcode/383-Ransom-Note/RandomNote.go
func canConstruct(ransomNote string, magazine string) bool {
	if len(ransomNote) > len(magazine) {
		return false
	}
	var count [26]int
	for _, val := range magazine {
		index := int(val) - int('a')
		count[index]++
	}
	for _, val := range ransomNote {
		index := int(val) - int('a')
		count[index]--
		if count[index] < 0 {
			return false
		}
	}
	return true
}
